docs/guides/cluster-to-advanced-cluster-migration-guide.md

```diff
@@ -0,0 +1,181 @@
+# Cluster to Advanced Cluster Migration Guide
+
+**Objective**: Guide users to replace the `mongodbatlas_cluster` resource with the `mongodbatlas_advanced_cluster` resource.
+
+**Note**: This guide focus on the resource migration as the data source migration is only requiring a resource_type change from `data.mongodbatlas_cluster` to `data.mongodbatlas_advanced_cluster`.  However, pay attention to the [output changes.](#output-changes)
+
+## Main Changes Between `mongodbatlas_cluster` and `mongodbatlas_advanced_cluster`
+
+Key differences in the configuration:
+
+1. Replication Spec Configuration: Support different node types (electable, analytics, read_only) where hardware configuration can differ between node types.
+2. Provider Settings: Moved from the top level to the replication spec allowing you to create multi-cloud clusters.
+3. Auto Scaling: Moved from the top level to the replication spec allowing you to scale replication specs individually.
+4. Backup Configuration: Renamed from `cloud_backup` to `backup_enabled`.
+
+### Example1: Old Configuration
+
+```terraform
+resource "mongodbatlas_cluster" "this" {
+    project_id                   = var.project_id
+    name                         = "legacy-cluster"
+    cluster_type                 = "REPLICASET"
+
+    # Provider Settings "block"
+    provider_instance_size_name = "M10" # 1
+    provider_name               = "AWS" # 2
+    
+    auto_scaling_disk_gb_enabled = true # 3
+    cloud_backup                 = true # 4
+
+    replication_specs {
+     num_shards = 1
+     regions_config {
+       region_name     = "US_EAST_1"
+       priority        = 7
+       electable_nodes = 3 # 1
+       analytics_nodes = 1 # 1
+       read_only_nodes = 0 # 1
+       }
+     }
+
+    }
+```
+
+### Example2: New Configuration
+
+```terraform
+    resource "mongodbatlas_advanced_cluster" "this" {
+    project_id             = var.project_id
+    name                   = "advanced-cluster"
+    cluster_type           = "REPLICASET"
+    backup_enabled         = true # 4
+
+    # Replication specs
+    replication_specs {
+      auto_scaling { # 3
+        disk_gb_enabled = true
+      }
+      region_configs {
+        region_name   = "US_EAST_1"
+        priority    = 7
+        provider_name = "AWS" # 2
+        
+        electable_specs { # 1
+            instance_size = "M10"
+            node_count    = 3
+        }
+        analytics_specs { # 1
+            instance_size = "M10"
+            node_count    = 1
+        }
+      }
+    }
+```
+
+### Output Changes
+
+- container_id:
+  - before: `mongodbatlas_cluster.this.replication_specs[0].container_id` was a flat string, e.g., `669644ae01bf814e3d25b963`
+  - now: `mongodbatlas_advanced_cluster.this.replication_specs[0].container_id` is a map, e.g., `{"AWS:US_EAST_1": "669644ae01bf814e3d25b963"}`
```

Comment:
stringfied JSON object instead of a map?

Reply:
What do you mean?